crates/services/miscellaneous/src/lib.rs

```diff
@@ -3041,28 +3041,31 @@ impl MiscellaneousService {
             }
         }
         for col_update in collection_updates.into_iter() {
-            let metadata::Model { id, .. } = self
```

Comment:
Why was this change made?

Reply:
Previously in anilist we were directly unwrapping regardless of if we received a proper response which is normally fine but if you get  rate limited will cause the error to unwrap all the way back to the background service causing some confusing errors. As part of this change we added a proper handler for if the anilist data is rate limited and this receives that error rather than passing it up the stack.

Reply:
This is namely an issue when the metadata is being grabbed during a background sync as the tasks following the metadata check wouldnt ever be handled because of the error never being handled. This could result in a non-zero amount of time that things which should be running arent running correctly.

Reply:
https://github.com/IgnisDa/ryot/blob/3b6b9226ce46493bdefd65e3bc50ad26a416fb45/crates/services/miscellaneous/src/lib.rs#L3052

In the event of a rate handle anything after this line would never have run due to the error never being received

Reply:
Makes sense. Thanks for the fix!